### ISSUE TITLE
Add documentation for [pipx.run] entry point

### ISSUE THREAD

**Issue by matthewfeickert** [pypa/pipx#1371]
> TODO: Write documentation on this.

_Originally posted by @uranusjr in https://github.com/pypa/pipx/issues/615#issuecomment-769569226_

I only learned about the `pipx.run` entry point earlier today by asking about it, but it would be quite useful to have some docs on this.

**Comment by chrysle** [comment]
I must admit I also wasn't aware of that entry point. Thanks for raising this issue!

**Comment by jaraco** [comment]
Looks like the implementation is here:

https://github.com/pypa/pipx/blob/dc9f9a849a918c68bc2cc99c72651022371fcf9f/src/pipx/venv.py#L405-L407

It also looks like it doesn't play well with local installs.

**Comment by chrysle** [comment]
> ~It also looks like it doesn't play well with local installs.~ (edit: wasn't correct on latest code)

Yes, I encountered this issue myself when trying out the entry point. The fix was included in version 1.6.0.

FYI mentioned this in my proposal for a new PyPUG guide on packaging command-line applications at pypa/packaging.python.org#1553; if anyone would care to take a look, that would be greatly appreciated!
